Entity backend

MojoOno opened the pull request:
This pull request contains the following:

- The creation of Entities; Assignement, Question, MathTeam + the modification of an already existing UserAccount entity
- Attributes + relations have been added to the entities above.
- Constructors and methods to Assignment has been added.
- - I am unsure whether the entities contain the correct constructors as i am having a hard time visualizing the actual usage + creation of said entities. Please let me know where changes are needed and they will be fixed asap.
- I have tried to make unique constraints to relevant attributes.
- Lombok has been used to make NoArgs, AllArgs constructors + Getters for all entities
- Relations according to the mock-up has been added to all entities
- - I have added a OneToMany relations between UserAccount(1) and MathTeam(*), as MathTeam according to the documentation have a UserAccount owner attribute.
- A few comments have been made in the Questions entity where i have changed danish variable names to english ones.
- I have kept the following entities; Hotel and Room for template usage.
- Added entities to HibernateConfig
- Added a constructor to Assignment taking a DTO as input.
- Changed method names to getters in Assignment

MojoOno committed:
Added field attributes to UserAccount

MojoOno committed:
Added fields attributes to Assignment

MojoOno committed:
added field attributes to MathTeam

MojoOno committed:
Added relation from UserAccount to MathTeam

MojoOno committed:
Added field attributes to Question

MojoOno committed:
added constructors to Question and uninque constraint to questionNumber

MojoOno committed:
added unique constraints to UserAccount

MojoOno committed:
removed whitespace and unused import

MojoOno committed:
Added constructors

MojoOno committed:
Added constructor and methods to Assignment

MojoOno committed:
added constructors to MathTeam

MojoOno committed:
Removed whitespace and obsolete comments

MojoOno committed:
Removed Hotel and Room from hibernateConfig and added MathTeah, Question and Assignment.

MojoOno committed:
Added a constructor with and AssignmentDTO as parameter. I am in doubt though wether it is correct or not.
changed method names to getters and coordinated names with Idris

AndyTheDragon commented on src/main/java/dat/entities/Assignment.java:
```diff
@@ -0,0 +1,59 @@
+    private String introText;
+
+    @ManyToOne
+    private MathTeam mathTeam;
```
tilføj venligst setter

AndyTheDragon commented on src/main/java/dat/entities/Assignment.java:
```diff
@@ -0,0 +1,59 @@
+    private MathTeam mathTeam;
+
+    @ManyToOne
+    private UserAccount owner;
```
tilføj setter

MojoOno committed:
added field attribute category

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,51 @@
+    private String description;
+
+    @OneToMany
+    private Set<Assignment> assignments;
```
tilføj addAssignment(Assignment assignment) metode

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,51 @@
+    private Set<Assignment> assignments;
+
+    @ManyToOne
+    private UserAccount owner;
```
tilføj setter

AndyTheDragon commented on src/main/java/dat/entities/Question.java:
```diff
@@ -0,0 +1,95 @@
+    private String pictureURL;
+
+    @ElementCollection
+    private List<String> level = new ArrayList<>(Arrays.asList("A", "B", "C", null));
```
Dette skal være en String eller en enum

MojoOno replied:
Har lavet det til en String level;

AndyTheDragon commented on src/main/java/dat/entities/Question.java:
```diff
@@ -0,0 +1,95 @@
+    private List<String> level = new ArrayList<>(Arrays.asList("A", "B", "C", null));
+
+    @ElementCollection
+    private List<Integer> testForm = new ArrayList<>(Arrays.asList(1, 2, null)); //Har kaldt denne for testForm istedet for delprøve. Den kan vi lige vende sammen.
```
lav en Enum TestFormat { MED, UDEN } og kald feltet her for testFormat af typen TestFormat.

AndyTheDragon commented on src/main/java/dat/entities/Question.java:
```diff
@@ -0,0 +1,95 @@
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    @Enumerated(EnumType.STRING)
+    private Set<Roles> roles = new HashSet<>();
```
question indeholder ikke Roles?

AndyTheDragon replied:
skal være String licens

AndyTheDragon commented on src/main/java/dat/entities/Question.java:
```diff
@@ -0,0 +1,95 @@
+    private Set<Roles> roles = new HashSet<>();
+
+    @ManyToMany
+    private Set<Assignment> assignments;
```
Tilføj addAssignment og removeAssignment

AndyTheDragon commented on src/main/java/dat/entities/Question.java:
```diff
@@ -0,0 +1,95 @@
+    private Set<Assignment> assignments;
+
+    @ManyToMany
+    private Set<MathTeam> mathTeams;
```
Tilføj addTeam og removeTeam()

MojoOno committed:
added field attribute license

AndyTheDragon commented on src/main/java/dat/entities/UserAccount.java:
```diff
@@ -20,23 +20,63 @@
+    private Set<Assignment> assignments = new HashSet<>();
+
+    @OneToMany
+    private Set<MathTeam> mathTeams = new HashSet<>();
```
tilføj addTeam(Team) og removeTeam(Team)

AndyTheDragon commented on src/main/java/dat/entities/Assignment.java:
```diff
@@ -0,0 +1,59 @@
+    private UserAccount owner;
+
+    @ManyToMany
+    private Set<Question> questions;
```
tilføj addQuestion(Question) og removeQuestion(Question)

MojoOno committed:
Added constructor with a MathTeamDTO as input

MojoOno committed:
Refactored level to a String.
Refactored testFormat to an ENUM and created the ENUM class TestFormat.

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,51 @@
+    private UserAccount owner;
+
+    @ManyToMany
+    private Set<Question> questions;
```
tilføj addQuestion(Question) og removeQuestion(Question)

MojoOno committed:
Added setters in Assignment

MojoOno committed:
Added add and removeAssignment method.
Added setters

MojoOno committed:
Removed comment from points.
Removed roles from Question.
Added constructor with DTO as input.
Added license to relevant constructors.
Added bi-directional methods to Question class.

MojoOno committed:
Added bi-directional methods to MathTeam

MojoOno committed:
Added bi-directional to Assignment

MojoOno committed:
Added bi-directional methods
Note the out-commented line in addQuestion

MojoOno committed:
Added a setter to Question for MathTeams, but am unsure if nescesary.

MojoOno committed:
added bi-directional

MojoOno commented:
Went through all comments and added functionality as requested.

I am still in doubt of some of the constructors, and hope we can look at this together.
The bi directionals also need a keen eye on them, as it's quite confusing at times.

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,102 @@
+        if (question != null)
+        {
+            questions.add(question);
+            //question.setMathTeams(this);
```
brokker sig fordi det skal være en add metode, question.addMathTeam(this)

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,102 @@
+        if (question != null)
+        {
+            questions.remove(question);
+            question.setMathTeams(null);
```
skal være remove metode

AndyTheDragon commented on src/main/java/dat/entities/MathTeam.java:
```diff
@@ -0,0 +1,102 @@
+        }
+    }
+
+    public void setAssignments(Set<Assignment> assignments)
```
skal være en add metode addAssignment(Assignment)

MojoOno committed:
changed add question to use proper method.
Vice versa with removeQuestion method.
Removed obsolete setter method.
